components/movementsensor/gpsutils/cachedData.go

```diff
@@ -88,6 +88,17 @@ func (g *CachedData) ParseAndUpdate(line string) error {
 	return g.nmeaData.ParseAndUpdate(line)
 }
 
+// GGA returns the GGA message of the sensor, or an error.
+func (g *CachedData) GGA() (string, error) {
+	g.mu.RLock()
+	defer g.mu.RUnlock()
+
+	if g.nmeaData.LastGGAMessage == "" {
```

Comment:
Note to myself: it's okay to do this because `g.nmeaData` is only ever modified (by calling `g.nmeaData.ParseAndUpdate` when `g.mu` is locked. So, there's no chance of a race condition where `LastGGAMessage` is being updated while we're trying to read it.